tests/e2e/lib/apps.go

```diff
@@ -449,105 +446,138 @@ func RunMustGather(oc_cli string, artifact_dir string) error {
 	return err
 }
 
-func makeRequest(request string, api string, todo string) {
-	params := url.Values{}
-	params.Add("description", todo)
-	body := strings.NewReader(params.Encode())
-	req, err := http.NewRequest(request, api, body)
-	if err != nil {
-		log.Printf("Error making post request to todo app before Prebackup  %s", err)
-	}
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		log.Printf("Response of todo POST REQUEST  %s", resp.Status)
-	}
-	defer resp.Body.Close()
-}
-
 // VerifyBackupRestoreData verifies if app ready before backup and after restore to compare data.
-func VerifyBackupRestoreData(clientv1 client.Client, artifact_dir string, namespace string, routeName string, app string, prebackupState bool, twoVol bool, backupRestoretype BackupRestoreType) error {
+func VerifyBackupRestoreData(ocClient client.Client, kubeClient *kubernetes.Clientset, kubeConfig *rest.Config, artifactDir string, namespace string, routeName string, serviceName string, app string, prebackupState bool, twoVol bool) error {
 	log.Printf("Verifying backup/restore data of %s", app)
-	appRoute := &routev1.Route{}
-	backupFile := artifact_dir + "/backup-data.txt"
-	routev1.AddToScheme(clientv1.Scheme())
-	err := clientv1.Get(context.Background(), client.ObjectKey{
-		Namespace: namespace,
-		Name:      routeName,
-	}, appRoute)
+	appEndpointURL, proxyPodParams, err := getAppEndpointURLAndProxyParams(ocClient, kubeClient, kubeConfig, namespace, serviceName, routeName)
 	if err != nil {
 		return err
 	}
-	appApi := "http://" + appRoute.Spec.Host
-	appEndpoint := appApi + "/todo-incomplete"
-	volumeEndpoint := appApi + "/log"
 
-	//if this is prebackstate = true, add items via makeRequest function. We only want to make request before backup
-	//and ignore post restore checks.
-	log.Printf("PrebackState: %t\n", prebackupState)
```

Comment:
would not remove this from logs, I think it is good for debugging them later

Reply:
I think that is really useless information in here. Every test consists of always two calls, one `PreBackupVerify` and another `PostRestoreVerify`, it's clear from the logs in which we are in without having additional `PrebackState` entry.

Here is log example. I think line first and last are duplications.
```
2024/01/25 20:46:02 Running pre-backup function for case mongo-csi-e2e
2024/01/25 20:46:02 Verifying backup/restore data of todolist
2024/01/25 20:46:02 PrebackState: true
```

Reply:
yeah, agreed (my concern is calling function with wrong bool, but that should be easy to catch as well without this log)

Reply:
That should really not happen, we are not setting the bool dynamically it's within Entry() that has true/false passed as a single value, so very limited place for mistake.

Example:
https://github.com/openshift/oadp-operator/blob/master/tests/e2e/backup_restore_suite_test.go#L321-L322